kong/clustering/data_plane.lua

```diff
@@ -62,11 +65,35 @@ function _M.new(parent)
 end
 
 
+local function sort(t)
```

Comment:
maybe we can have a basic performance on this sort function.
for example if we have a json with 5M of JSON, how long does it take to sort it?

Reply:
I can do it, but I would expect it to be minimal. Like not even close to all that schema validation etc. stuff that also happens there.

Reply:
Here some results with 68MB JSON file:

```
(feat/lax-version-checks *) bungle@pro:~/Sources/kong$ ll config.json
-rw-r--r--  1 bungle  staff    68M Jun 23 10:45 config.json
(feat/lax-version-checks *) bungle@pro:~/Sources/kong$ resty perf.lua 
7e111baf0b58b2362b6646d29ca5c458
took: 2.3029999732971 secs
```

And here is the `perf.lua`:
```lua
local cjson = require "cjson"
local tablex = require "pl.tablex"
local concat = table.concat
local tostring = tostring
local ngx_null = ngx.null
local md5 = ngx.md5
local open = io.open


local function sort(t)
  if t == ngx_null then
    return "/null/"
  end

  local typ = type(t)
  if typ == "table" then
    local i = 1
    local o = { "{" }
    for k, v in tablex.sort(t) do
      o[i+1] = sort(k)
      o[i+2] = ":"
      o[i+3] = sort(v)
      o[i+4] = ";"
      i=i+4
    end
    if i == 1 then
      i = i + 1
    end
    o[i] = "}"

    return concat(o, nil, 1, i)

  elseif typ == "string" then
    return '$' .. t .. '$'

  elseif typ == "number" then
    return '#' .. tostring(t) .. '#'

  elseif typ == "boolean" then
    return '?' .. tostring(t) .. '?'

  else
    return '(' .. tostring(t) .. ')'
  end
end


local function hash(t)
  return md5(sort(t))
end


local function read_file(path)
    local file = open(path, "rb")
    local content = file:read "*a"
    file:close()
    return content
end


local content = read_file("config.json")
local json = cjson.decode(content)


ngx.update_time()
local started = ngx.now()
print(hash(json))
ngx.update_time()
local ended = ngx.now()
print("took: ", ended - started, " secs")
```

Reply:
With 5MB `json` it takes on my `2010` Mac: 0.2 secs.

Reply:
Awesome!